### DISCUSSION TITLE
How to build clair v4

### DISCUSSION THREAD

**Discussion by leiqi96** [quay/clair#1647] [Q&A] [answered]
I can not find steps to build clair from [Makefile ](https://github.com/quay/clair/blob/main/Makefile)and [clair doc](https://quay.github.io/clair/reference/clairctl.html).

**Comment by hdonnay** [comment, marked as answer]
What did you try? `go build` should work just fine.

**Reply by leiqi96** [reply]
I want to build all clair compents from Makefile.
If there is no target to compile clair in Makefile，then what is the Makefile  for?

When quay released the latest version, they just use `go build`

**Reply by hdonnay** [reply]
Yeah, builds just use `go build`. You can see how the release container is built with the `Dockerfile`. The `Makefile` is for various tasks that have multiple steps or undiscoverable arguments.
